### ISSUE TITLE
Can't use iohook when deploying to heroku ( and yes I have installed the libraries)

### ISSUE THREAD

**Issue by dawoodsa** [wilix-team/iohook#293]
I am using iohook in my nodejs application which works fine when running the application locally. When running a build to heroku it is successful except that when I try opening the application it gives me an error as following:

`on_library_load [458]: XOpenDisplay failure!
app[web.1]: load_input_helper [1918]: Using XFree86 
keyboard layout.
 app[web.1]: load_input_helper [1920]: XKB support is required to accurately determine keyboard characters!
 app[web.1]: Segmentation fault`

I have also installed the libraries:
libx11-dev 
libxtst-dev 
libxt-dev 
libx11-xcb-dev 
libxkbcommon-dev 
libxkbcommon-x11-0

I think the problem is from the Xserver(users machine), not the XClient(the heroku server which recieves the data)
but I am very confused as to what is problem because when running the app locally it works fine.

My environment:
Heroku stack: 18(runs ubuntu),
Nodejs : v12.18.3,
My local operating system : windows 10,

Any help is greatly appreciated.

**Comment by WilixLead** [comment]
Hello @dawoodsa. As I know, Heroku - is a container like service or VPS without connected video driver. So system libs that used in X server can't find any video card and kernel just don't init required modules (or this modules just not installed to kernel)
So, if you checking iohook locally with video card - it is not the same with heroku. 
BTW: Why you need iohook in Heroku without physical keyboard and mouse? (may be I don't know something)

**Comment by dawoodsa** [comment]
Thank you for your reply , but I'm a little confused; I get that Heroku servers aren't connected to a video driver, but when referring to the X server, do you mean heroku, or the user that is connected to the nodejs application. And what are the required modules (I am lost). As for your question, I am using iohook to listen for the users keyboard events outside the scope of the browser.

**Comment by dawoodsa** [comment]
Thank you @WilixLead for your reply. I am having a little confusion; I get that Heroku isn't connected to a video driver, but when referring to the X server do you mean heroku, or the user that is connecting the nodejs application. And what are the required modules I shouldn't init (I am truly lost). As for your question, I am using iohook to register keyboard events outside the scope of a browser from users that connect to my nodejs web application.

**Comment by WilixLead** [comment]
@dawoodsa This module work on desktop side, not on browser. If you need browser handler - you should use browser API like addEventHandler or just any npm module for frontend side.
This module can handle only system events, that emits from local computer (where this module should be used).
You can't handle remote events by connecting via API or HTML page, iohook should work on client side (on desktops) only

**Comment by dawoodsa** [comment]
@WilixLead ,
` const io = req.app.get('socket-io');
  io.on('connection', socket => {
    console.log('i am the server ! and socket io is connected')
    ioHook.on('keydown',e => {
      socket.emit('keydown',e);
    })
    ioHook.on('keyup',e => {
      socket.emit('keyup',e);
    })
    ioHook.stop();
  });`
I used socket io to send a event to the client-side when the iohook event is regestered.
And it worked just fine, when I deployed to heroku I can't use it.
I think what you are saying is it is made for electron only, so the modules must be installed on the device that is using the application, unlike a webapplication where the modules are installed on the server not the localmachine.
